walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,118 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
 
+/**
+ * This class only contains the query used to find the time ranges when a
+ * meeting request coincides with available time.
+ */
 public final class FindMeetingQuery {
+  // This comparator is used to sort a list of events by start time
+  // in ascending order.
+  private Comparator<Event> eventComparator = new Comparator<Event>() {
+    @Override
+    public int compare(Event event1, Event event2) {
+      return event1.getWhen().start() - event2.getWhen().start();
+    }
+  };
+  /**
+   * The way the query works is by going through the sorted events, and
+   * if the current event does not have any of the people in the meeting
+   * request, the event is skipped. If it does share people with the request,
+   * availableTimeDuration is set to the available time from the end of the
+   * previous event until the start of the current event. Then, if
+   * availableTimeDuration is at least the same size as the duration of the
+   * request, a TimeRange from pastEventEnd to currentEventStart is
+   * added to the list of available times.
+   *
+   * Time complexity: 
+   * n: number of events
+   * m: number of attendees in every event
+   * If m > logn, then O(nm)
```

Comment:
Always assume worst case for big-O analysis. So, this would simply be O(nm)

Reply:
Done